File: app/src/main/java/com/annapol04/munchkin/network/PlayClientDummy.java
package com.annapol04.munchkin.network;

import android.app.Activity;
import android.content.Intent;

import com.annapol04.munchkin.engine.PlayClient;

public class PlayClientDummy extends PlayClient {
    private boolean loggedIn = false;

    @Override
    public void setActivity(Activity activity) {

    }

    @Override
    public boolean isLoggedIn() {
        return loggedIn;
    }

    @Override
    public void login() {
        loggedIn = true;

        changeMatchState(MatchState.LOGGED_IN);
    }

    @Override
    public void processActivityResults(int requestCode, int resultCode, Intent data) {

    }

    @Override
    public void startQuickGame() {
        changeMatchState(MatchState.STARTED);
    }

    @Override
    public void sendToAll(byte[] message) {
        messageReceived(message);
    }

    @Override
    public int getAmountOfPlayers() {
        return 2;
    }
}

File: app/src/main/java/com/annapol04/munchkin/network/PlayClientDummy.kt
package com.annapol04.munchkin.network

import android.app.Activity
import android.content.Intent

import com.annapol04.munchkin.engine.PlayClient

class PlayClientDummy : PlayClient() {
    override var isLoggedIn = false
        private set

    override val amountOfPlayers: Int
        get() = 3

    override fun setActivity(activity: Activity?) {

    }

    override fun login() {
        matchState = ClientState.LOGGED_IN
    }

    override fun processActivityResults(requestCode: Int, resultCode: Int, data: Intent?) {

    }

    override fun startQuickGame() {
        matchState = ClientState.STARTED
    }

    override fun leaveGame() {

    }

    override fun sendToAll(message: ByteArray) {
        messageReceived(message)
    }
}
